cou/utils/openstack.py

```diff
@@ -125,6 +125,97 @@
     ]
 )
 
+LTS_SERIES = {
+    "bionic": "queens",
+    "focal": "ussuri",
+    "jammy": "yoga",
+}
+
+
+def add_charms(mapping: dict) -> dict:
+    """Add charms based on CHARM_TYPES.
+
+    :param mapping: Dictionary containing the services.
+    :type mapping: dict
+    :return:  Dictionary containing the services and charms
+    :rtype: dict
+    """
+    mapping_charms = mapping.copy()
+    ubuntu_series = mapping_charms.keys()
+    for series in ubuntu_series:
+        for charm_type, charms in CHARM_TYPES.items():
+            for charm in charms:
+                if charm_type in mapping_charms[series].keys():
+                    mapping_charms[series][charm] = mapping_charms[series][charm_type]
```

Comment:
Having heavily nested "for" loops is not a good practice. We should discuss and come up with some  ways to simplify the logic

Reply:
what you think about something like this?

```python
def add_charms(mapping: dict) -> dict:
    mapping_charms = mapping.copy()
    ubuntu_series = mapping_charms.keys()
    for series in ubuntu_series:
        for charm_type, charms in CHARM_TYPES.items():
            new_mapping = {
                charm: mapping_charms[series][charm_type]
                for charm in charms
                if charm_type in mapping_charms[series].keys()
            }
            mapping_charms[series].update(new_mapping)
    return mapping_charms
```

In the end of the day it's 3 for loops but it will feel like 2.

Reply:
what you think about something like this?

```python
def add_charms(mapping: dict) -> dict:
    mapping_charms = mapping.copy()
    ubuntu_series = mapping_charms.keys()
    for series in ubuntu_series:
        for charm_type, charms in CHARM_TYPES.items():
            new_mapping = {
                charm: mapping_charms[series][charm_type]
                for charm in charms
                if charm_type in mapping_charms[series].keys()
            }
            mapping_charms[series].update(new_mapping)
    return mapping_charms
```

In the end of the day it's 3 for loops but it will feel like 2.

Reply:
based on our conversation on the cou-sync, I've created a new parameter `family` during the creation of the application, so we can get rid of this function

Reply:
I could not see the `family` parameter? Will you pushed new version?